big.js

```diff
@@ -111,7 +111,16 @@
 
       // Retain a reference to this Big constructor.
       // Shadow Big.prototype.constructor which points to Object.
-      x.constructor = Big;
+      if (typeof Object.defineProperty === 'function') {
+        // If the Object prototype is frozen, the "constructor" property is non-writable. This means that any objects which inherit this
+        // property cannot have the property changed using an assignment. If using strict mode, attempting that will cause an error. If not
+        // using strict mode, attempting that will be silently ignored.
+        // However, we can still explicitly shadow the prototype's non-writable property by defining a new property on this object.
+        Object.defineProperty(x, 'constructor', { value: Big });
+      } else {
+        // If Object.defineProperty() doesn't exist, attempt to shadow this property using the assignment operator.
+        x.constructor = Big;
+      }
```

Comment:
I noticed that this package is designed to only use ECMAScript 3, so it works on all browsers.

I looked in the ECMAScript 3 specification and did not find mention of `Object.defineProperty()` or `Object.freeze()`.

I'm not sure exactly when these global functions were introduced in the ECMAScript specification, but here is are the MDN compatibility pages for both of them:

* https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty#browser_compatibility
* https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze#browser_compatibility

So, `Object.defineProperty()` was supported _before_ `Object.freeze()`. This means we can assume that if `Object.defineProperty()` does not exist, then the global Object prototype has not been frozen either.

I think the safest thing to do here is to check if `Object.defineProperty()` exists and use it, otherwise fall back to shadowing these properties using the assignment operator.